src/jws.rs

```diff
@@ -72,6 +72,7 @@ pub fn sign_bytes(algorithm: Algorithm, data: &[u8], key: &JWK) -> Result<Vec<u8
     let signature = match &key.params {
         #[cfg(feature = "ring")]
         JWKParams::RSA(rsa_params) => {
+            rsa_params.validate_key_size()?;
```

Comment:
Maybe you could do this check at creation time (so would require a constructor)

Reply:
The 2048-bit minimum is for use with the algorithms specified in RFC 7518 rather than for the key representation in general. So probably this validation function should be defined separately in `jws.rs` rather that on RSAParams.
We might encounter smaller RSA keys - in `did:webkey`, for example - although would not be able to do much with them. (Alternatively, we could drop such keys).